Sources/HaCWebsiteLib/ViewModels/Script.swift

```diff
@@ -0,0 +1,68 @@
+import HaCTML
+import DotEnv
+import Foundation
+
+protocol JavaScriptable {
+  var javaScript: String {get}
+}
+
+extension String: JavaScriptable {
+  var javaScript : String {
+    // TODO: PROPERLY ESCAPE THIS! (eg. conver newlines to backslash)
+    // NOTE: A JSON LIBRARY COULD BE USED FOR THIS!
+    return "\"\(self)\""
+  }
+}
+
+extension URL: JavaScriptable {
+  var javaScript : String {
+    return "\(self)".javaScript
+  }
+}
+
+extension Date: JavaScriptable {
+  var javaScript : String {
+    let calendar = Calendar.current
+    let year = calendar.component(.year, from: self)
+    // We have to subtract 1 from the month as JavaScript months count from 0
+    // See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date
+    let month = calendar.component(.month, from: self) - 1
+    let day = calendar.component(.day, from: self)
+    let hour = calendar.component(.hour, from: self)
+    let minute = calendar.component(.minute, from: self)
+    let second = calendar.component(.second, from: self)
+    let milliseconds = 0
+    return "new Date(Date.UTC(\(year),\(month),\(day),\(hour),\(minute),\(second),\(milliseconds)))"
+  }
+}
+
+struct UnsafeRawJavaScript {
+   let rawScript : String
+   var javaScript : String {
+     return rawScript
+   }
+}
+
+/**
+ * This class is used in order load front-end scripts from a file relative to the current path for browsing pleasure.
+ *
+ * It depends on the build system loading files into the 'Data' folder before running.
+ */
+struct Script : Nodeable {
+  let file : String
+  let escapes : [String: JavaScriptable]
+  let directory : String = DotEnv.get("BACKEND_JS_DIR")!
```

Comment:
Why is this an environment variable? We're committing JS files to the repo in a specific place, so it makes sense to have this in code as a constant, right?

Reply:
That is because I thought I was copying existing patterns, like the DATA_DIR definition:
```
DATA_DIR: /root/hac-website/Data
BACKEND_JS_DIR: /root/hac-website/Sources/HaCWebsiteLib/ViewModels
```

Reply:
I'm not particularly committed to it. It's something I did quite early before I properly understood the structure of the project.

Reply:
In the case of the `DATA_DIR`, there is no dependency on the files in our project.

In every environment we have, the JS files are going to be in the same place (as they are committed). As it's not environment dependent, does it make sense for the environment variable to own it?

Reply:
No it doesn't, will change it.

Reply:
Another reason this is problematic is we have to set it everywhere it's important. For example we also have to set it on any Heroku instances we have running.